Modules/Loadable/Markups/Testing/Python/MarkupsGenericStorageTest.py

```diff
@@ -0,0 +1,178 @@
+    parent.contributors = ["Johan Andruejol (Kitware Inc."]
+    parent.helpText = """
+    This is a test for the Generic Markups Storage NOde
+    """
+    parent.acknowledgementText = """
+    This file was originally developed by Johan Andruejol and was partially funded by Allen Institute".
+""" # replace with organization, grant and thanks.
+    self.parent = parent
+
+    # Add this test to the SelfTest module's list for discovery when the module
+    # is created.  Since this module may be discovered before SelfTests itself,
+    # create the list if it doesn't already exist.
+    try:
+      slicer.selfTests
+    except AttributeError:
+      slicer.selfTests = {}
+    slicer.selfTests['MarkupsGenericStorageTest'] = self.runTest
+
+  def runTest(self):
+    tester = MarkupsGenericStorageTestTest()
+    tester.runTest()
+
+#
+# qMarkupsGenericStorageTestWidget
+#
+
+class MarkupsGenericStorageTestWidget(ScriptedLoadableModuleWidget):
+
+  def setup(self):
+    ScriptedLoadableModuleWidget.setup(self)
+
+#
+# MarkupsGenericStorageTestLogic
+#
+
+class MarkupsGenericStorageTestLogic(ScriptedLoadableModuleLogic):
+
+  def __init__(self):
+    pass
+
+#
+# MarkupsGenericStorageTestTest
+#
+
+class MarkupsGenericStorageTestTest(ScriptedLoadableModuleTest):
+
+  def setUp(self):
+    """ Reset the state for testing.
+    """
+    slicer.mrmlScene.Clear(0)
+
+  def runTest(self):
+    """Run as few or as many tests as needed here.
+    """
+    self.setUp()
+    self.test_MarkupsGenericStorageTest1()
+    self.test_MarkupsGenericStorageTest2()
+
+  def saveAndReload(self, markup):
+    self.assertIsNone(markup.GetStorageNode())
+
+    self.assertTrue(markup.AddDefaultStorageNode())
+    storage = markup.GetStorageNode()
+    self.assertIsNotNone(storage)
+    self.assertTrue(storage.IsA('vtkMRMLMarkupsGenericStorageNode'))
+
+    tempFile = qt.QTemporaryFile("test-MarkupsGenericStorageTest-XXXXXX.json")
+    self.assertTrue(tempFile.open())
+
+    slicer.util.saveNode(markup, tempFile.fileName())
+    slicer.mrmlScene.Clear(0)
+
+    # Make sure it can be read by python json
+    file = open(tempFile.fileName())
+    data = json.load(file)
+    markupsKeys = [
+      'Locked',
+      'MarkupLabelFormat',
+      'TextList',
+      'TextList_Size',
+      'Markups',
+      'Markups_Size',
+      ]
+    for markupsKey in markupsKeys:
+      self.assertTrue(markupsKey in data.keys(), msg="Checking for key %s" %markupsKey)
+
+    for i in range(int(data['Markups_Size'])):
+      markupSubKeys = [
+        'ID',
+        'Label',
+        'Description',
+        'AssociatedNodeID',
+        'Points',
+        'OrientationWXYZ',
+        'Selected',
+        'Locked',
+        'Visibility',
+        ]
+      for subKey in markupSubKeys:
+        self.assertTrue(subKey in data['Markups'][i].keys(), msg="Checking for sub key %s" %subKey)
+
+    loaded = slicer.util.loadNodeFromFile(tempFile.fileName(), 'Markups', returnNode=True)
```

Comment:
would be nice to have `slicer.util.loadMarkups` or `slicer.util.loadMarkupsList`